cloudinit/distros/debian.py

```diff
@@ -220,6 +221,10 @@ def set_keymap(self, layout: str, model: str, variant: str, options: str):
         # be needed
         self.manage_service("restart", "console-setup")
 
+    def wait_for_network(self):
+        """Ensure that cloud-init's network service has network connectivity"""
+        netplan.wait_for_network()
```

Comment:
This looks like this approach will work for broader environments than just netplan and just Ubuntu and/or Debian. I like that you are keeping it specific for the time being to environments which have netplan.

Don't we also want to confirm `netplan.available()` before calling `netplan.wait_for_network()` to avoid calling this on debian environments which don't have netplan.io installed?

This makes sense if the debian image contains netplan, but what happens on Debian images when they don't have have netplan?

Reply:
Good point. I was assuming it was fine for debian because of the netplan usage there too.
I think instead I should move this method into ubuntu.py. If netplan *isn't* available, we have nothing left to wait on. On Ubuntu, we can assume we have netplan available for supported images. If somebody goes off the beaten path in their image there's really no "right" thing to do other than log an error.